plugins/modules/git_config.py

```diff
@@ -197,94 +217,127 @@ def main():
     # Set the locale to C to ensure consistent messages.
     module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C', LC_CTYPE='C')
 
-    if params['name']:
-        name = params['name']
-    else:
-        name = None
+    name = params.get('name', "")
```

Comment:
`'name' in params` is always `True`. So you basically write
```suggestion
    name = params['name']
```
here. Also you are changing semantics here, before `""` was changed to `None`, now it is kept as `""`. Is that intentional? Did you change other code so that the behavior does not change?

Reply:
I changed from `None` to empty string, because within this python file for if checks, it stays the same but for string concatinations it is easier (because no check must be done).

Reply:
In that case you need to write `name = params['name'] or ''`, otherwise the value is sometimes `None` and sometimes `''`. The original code basically did `name = params['name'] or None` to make sure that it cannot be `''`.

Reply:
Ah thx. Good to know. I've updated it accordingly.